### PR TITLE
fix: questions-ui.yml missing latest configuration.

### PR DESCRIPTION
**Pull request by jvanz**
## Description

The questions-ui.yml file is missing the latest configuration added: ignoreValues. This commits adds this new field in the file. As well as, configures the other fields to be hidden when user sets ignoreValue to false.

### CODE REVIEW

**Comment by jvanz** [comment]
@kravciak can you review this file as well? :)

**Review by viccuad — APPROVED** [review]
LGTM

**Comment by kravciak** [comment]
> @kravciak @aalves08 can you review this file as well? :)

taking a look now, sorry for the delay..

File: artifacthub-pkg.yml [hunk 1]
```diff
@@ -89,27 +101,39 @@ annotations:
...
       subquestions:
+        - default: false
+          group: Settings
+          label: Ignore values
+          title: Ignore values
+          type: boolean
+          variable: memory.ignoreValues
+          tooltip: >-
+            If set true, skips the enforcement of specific values and only enforces
+            that requests and limits are set
```
Comment (kravciak) [inline artifacthub-pkg.yml:112]:
```suggestion
```

File: artifacthub-pkg.yml [hunk 2]
```diff
@@ -60,27 +60,39 @@ annotations:
...
       subquestions:
+        - default: false
+          group: Settings
+          label: Ignore values
+          title: Ignore values
+          type: boolean
+          variable: cpu.ignoreValues
+          tooltip: >-
+            If set true, skips the enforcement of specific values and only enforces
+            that requests and limits are set
```
Comment (kravciak) [inline artifacthub-pkg.yml:71]:
```suggestion
```

File: questions-ui.yml [hunk 3]
```diff
@@ -17,27 +17,39 @@ questions:
...
   variable: cpu
   subquestions:
+    - default: false
+      group: Settings
+      label: Ignore values
+      title: Ignore values
+      type: boolean
+      variable: cpu.ignoreValues
+      tooltip: >-
+        If set true, skips the enforcement of specific values and only enforces
```
Comment (kravciak) [inline questions-ui.yml:27]:
```suggestion
       Skip enforcement of specific values, but ensure that requests and limits are set
```

File: questions-ui.yml [hunk 4]
```diff
@@ -17,27 +17,39 @@ questions:
...
+        that requests and limits are set
```
Comment (kravciak) [inline questions-ui.yml:28]:
```suggestion
```

File: questions-ui.yml [hunk 5]
```diff
@@ -46,27 +58,39 @@ questions:
...
   variable: memory
   subquestions:
+    - default: false
+      group: Settings
+      label: Ignore values
+      title: Ignore values
+      type: boolean
+      variable: memory.ignoreValues
+      tooltip: >-
+        If set true, skips the enforcement of specific values and only enforces
```
Comment (kravciak) [inline questions-ui.yml:68]:
```suggestion
        Skip enforcement of specific values, but ensure that requests and limits are set
```

File: questions-ui.yml [hunk 6]
```diff
@@ -46,27 +58,39 @@ questions:
...
+        that requests and limits are set
```
Comment (kravciak) [inline questions-ui.yml:69]:
```suggestion
```

**Review by kravciak — CHANGES_REQUESTED** [review]
I added suggestion for info message (ignore if you prefer yours), but I found an issue:

- I create policy initially with some values
- I edit policy and select `Ignore values`

UI does not clean hidden inputs when I select checkbox, so it produces yaml:
```yaml
  settings:
    cpu:
      ignoreValues: true
    memory:
      defaultLimit: '100'
      defaultRequest: '100'
      ignoreValues: true
      maxLimit: '100'
```

Which results in `Values cannot be true when any quantities are defined` error:
```
2024-04-11T18:54:06.256680Z  INFO validate_settings{self=PolicyEvaluator { runtime: "wapc" } settings={"allowedHostPaths": Array [Object {"pathPrefix": String("/tmp"), "readOnly": Bool(true)}]}}:policy_log{self=EvaluationContext { policy_id: "clusterwide-do-not-share-host-paths", callback_channel: Some(...), allowed_kubernetes_resources: {} }}: policy_log: accepting settings data={}
Error: [clusterwide-crs] settings are not valid: Some("Provided settings are not valid: invalid memory settings\nignoreValues cannot be true when any quantities are defined")
```

**Comment by jvanz** [comment]
@kravciak I just updated this PR with the [change](https://github.com/kubewarden/container-resources-policy/pull/31) in the policy to allow the quantities together with the ignoreValues. Can you test it again?

**Comment by kravciak** [comment]
I set both CPU & Memory to ignoreValues and I can still create pod without requests section.
```
apiVersion: v1
kind: Pod
metadata:
  name: frontend
spec:
  containers:
  - name: app
    image: nginx:alpine
    resources:
      limits:
        memory: "128Mi"
        cpu: 100m
```
I found this in kubernetes doc:
```
# https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
Note: If you specify a limit for a resource, but do not specify any request, and no admission-time mechanism has applied a default request for that resource, then Kubernetes copies the limit you specified and uses it as the requested value for the resource.
```
It's a bit misleading considering that in README we say `enforce that requests and limits are set`  - only limits is enforced, requests is auto-completed by kubernetes.

Otherwise it works fine.

**Comment by jvanz** [comment]
> It's a bit misleading considering that in README we say `enforce that requests and limits are set` - only limits is enforced, requests is auto-completed by kubernetes.

Yeah, this can be misleading indeed. To help clarify that, I've added a note in the README to explain to the user that others controllers (e.g. kubernetes ones) can be mutating the resource before the policy. Which is the case that you reported. I decided to not change the original sentence that you highlighted because that will happen depending of the admission controller configuration.  Are you fine with that change?

**Review by kravciak — APPROVED** [review]

**Pull request merged**